### ISSUE TITLE
getOrCreate() method

### ISSUE THREAD

**Issue by rstacruz** [isaacs/node-lru-cache#102]
This seems like a very common use case, and in fact it's the preferred way you use a cache in Rails. Is there any reason it's not implemented?

```
cache.getOrCreate(key, () => { expensive computation })
```

Where:

```js
LRU.prototype.fetch = function (key, fn) {
  if (this.has(key)) return this.get(key)
  var result = fn()
  this.set(key, result)
  return result
}
```

**Comment by rstacruz** [comment]
(can't edit on mobile, but that should be .Has and not .Peek :) )

**Comment by airliti** [comment]
Not implementing a .getOrCreate method could even be a possible issue ( and as demonstrated with your implementation )! The timing between the `#.has` and `#.get` method could be just surpassing the `maxAge` resulting in an `undefined` value. I did an implementation for an LruCache for myself and a possible solution can be found through visiting https://github.com/airliti/estructura/blob/master/src/lib/LruCache.js#L457.

If your interested, I've got sample code to prove the problem.
